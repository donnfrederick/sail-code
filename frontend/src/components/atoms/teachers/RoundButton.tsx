import * as React from 'react'
import styled from 'styled-components'

interface Props {
  text: string
  size?: number
  fontSize?: number
  isSelected?: boolean
  alwaysOn?: boolean
  onClick?: (event: any) => void
}

export default (props: Props) => {
  const {
    text,
    size = 240,
    fontSize = 48,
    isSelected,
    alwaysOn = true,
    onClick
  } = props

  return (
    <Button
      size={size}
      fontSize={fontSize}
      data-selected={isSelected}
      alwaysOn={alwaysOn}
      onClick={onClick}
    >
      {text}
    </Button>
  )
}

const Button = styled<Props, any>('button')`
  appearance: none;
  display: block;
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  box-sizing: border-box;
  padding: 0;
  border: 4px solid #138efd;
  border-radius: 50%;
  outline: none;
  background-color: transparent;
  font-size: ${props => props.fontSize}px;
  font-weight: 500;
  letter-spacing: 0px;
  text-align: center;
  color: #138efd;

  &[data-selected='true'] {
    pointer-events: ${props => (props.alwaysOn ? 'auto' : 'none')};
    border: none;
    background-image: linear-gradient(315deg, #2eb1ff, #138efd);
    color: #ffffff;
  }
`
