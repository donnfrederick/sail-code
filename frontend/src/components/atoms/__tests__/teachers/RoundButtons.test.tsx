import RoundButton from 'components/atoms/teachers/RoundButton'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <Button />', () => {
  const tree = renderer.create(<RoundButton text="女性" />).toJSON()
  expect(tree).toMatchSnapshot()
})
