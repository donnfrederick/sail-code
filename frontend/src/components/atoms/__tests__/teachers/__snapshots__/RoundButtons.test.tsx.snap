// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Button /> 1`] = `
<button
  className="sc-bdVaJa feofRl"
  fontSize={48}
  size={240}
>
  女性
</button>
`;
